src/app/shared/services/sorting.service.ts

```diff
@@ -9,26 +10,29 @@ export class SortingService {
     constructor() {
     }
 
-    public handleSortValue(data): void {
-        this.sortValue = data;
+    public handleSortValue(sortValue: SortValueInterface): void {
+        this.sortValue = sortValue;
     }
 
     public sortProducts(products: ProductInterface[]): ProductInterface[] {
         if (!this.sortValue) {
             return products;
         }
-        const value = this.sortValue.sortBy.value;
-        if (this.sortValue.sortBy.direction === 'asc') {
-            return this.sortByAsc(products, value);
+        const value: ValueTypeEnum = this.sortValue.sortBy.value;
+        if (value in ValueTypeEnum) {
```

Comment:
Why do you need this check? `(value in ValueTypeEnum)`
If you remove this, nothing will change I suppose?

Reply:
Removed